queuelib/queue.py

```diff
@@ -88,6 +94,20 @@ def pop(self):
         self.info['tail'] = [tnum, tcnt, toffset]
         return data
 
+    def peek(self):
+        tnum, tcnt, toffset = self.info['tail']
+        if [tnum, tcnt] >= self.info['head']:
+            return
+        tfd = self.tailf.fileno()
+        tfd_initial_pos = os.lseek(tfd, 0, os.SEEK_CUR)
+        szhdr = os.read(tfd, self.szhdr_size)
+        if not szhdr:
+            return
```

Comment:
I wonder why we check the output of `os.read` here but not a few lines below.

I’m not 100% sure, but maybe it would make sense to expect the file _not_ to be corrupt, and let code fail naturally if it is corrupted.

Reply:
I "mocked" (in quotes because I didn't actually use `unittest.mock`) one of the underlying files and that allowed me to reach the affected lines: 4db96d0b0c5aaf2e1ace8144fd248e81a1afe9ae

Reply:
My point was that the likelihood of this breaking and the `os.read` below breaking is not that different, but here we check, and below we don’t. It feels inconsistent.

Should we have this below as well for consistency?

```
data = os.read(tfd, size)
if not data:
    return
```

The result would be slightly different, causing the scenario of `tfd` being empty to return `None` instead of `b''`.

However, my point was that `if not szhdr:` seems to only matter if something unexpected happened to the file, out of the control of queuelib. Is that something we should worry about? Can’t we just let things break loudly if something out of the control of queuelib corrupts the queue file?

In other words, instead of mocking to cover those lines with tests, wouldn’t it be OK to remove the `if not szhdr:` part altogether?

Reply:
Ok, I see what you mean. I think it's a valid concern, but to be honest, I don't know for sure why the check was included in the first place. I tend to agree that we should probably fail loudly, but I don't feel confident enough to fully support that without a deeper understanding of this part of the code. That said, I don't think it should be a blocker for this PR. I think it's consistent to do it in `peek` if we do it in `pop`, and we can open a new issue to keep track of it separately.